kmir/src/kmir/kmir.py

```diff
@@ -66,70 +67,94 @@ def kcfg_explore(self, label: str | None = None) -> Iterator[KCFGExplore]:
 
     def make_call_config(
-        self, parsed_smir: KApply, smir_info: SMIRInfo, start_symbol: str = 'main', sort: str = 'GeneratedTopCell'
+        self, smir_info: SMIRInfo, start_symbol: str = 'main', sort: str = 'GeneratedTopCell', init: bool = False
     ) -> tuple[KInner, list[KInner]]:
-
         if not start_symbol in smir_info.function_tys:
             raise KeyError(f'{start_symbol} not found in program')
 
-        _sym, _allocs, functions, items, types, _ = parsed_smir.args
-
         args_info = smir_info.function_arguments[start_symbol]
-
         locals, constraints = symbolic_locals(smir_info, args_info)
+        types, adts = self._make_type_and_adt_maps(smir_info)
 
-        subst = {
+        _subst = {
             'K_CELL': mk_call_terminator(smir_info.function_tys[start_symbol], len(args_info)),
             'STARTSYMBOL_CELL': KApply('symbol(_)_LIB_Symbol_String', (token(start_symbol),)),
             'STACK_CELL': list_empty(),  # FIXME see #560, problems matching a symbolic stack
             'LOCALS_CELL': list_of(locals),
-            'FUNCTIONS_CELL': KApply('mkFunctionMap', (functions, items)),
-            'TYPES_CELL': KApply('mkTypeMap', (map_empty(), types)),
-            'ADTTOTY_CELL': KApply('mkAdtMap', (map_empty(), types)),
+            'FUNCTIONS_CELL': self._make_function_map(smir_info),
+            'TYPES_CELL': types,
+            'ADTTOTY_CELL': adts,
         }
 
-        config = self.definition.empty_config(KSort(sort))
+        _init_subst: dict[str, KInner] = {}
+        if init:
+            _subst['LOCALS_CELL'] = list_empty()
+            init_config = self.definition.init_config(KSort(sort))
+            _, _init_subst = split_config_from(init_config)
 
-        return (Subst(subst).apply(config), constraints)
+        subst = Subst(_init_subst).compose(Subst(_subst))
 
-    def run_parsed(self, parsed_smir: KInner, start_symbol: KInner | str = 'main', depth: int | None = None) -> Pattern:
-        init_config = self.make_init_config(parsed_smir, start_symbol)
-        init_kore = self.kast_to_kore(init_config, KSort('GeneratedTopCell'))
-        result = self.run_pattern(init_kore, depth=depth)
-
-        return result
+        config = self.definition.empty_config(KSort(sort))
+        return (subst.apply(config), constraints)
 
-    def run_call(
-        self, parsed_smir: KApply, smir_json: SMIRInfo, start_symbol: str = 'main', depth: int | None = None
-    ) -> Pattern:
-        init_config, _ = self.make_call_config(parsed_smir, smir_json, start_symbol)
+    def run_smir(self, smir_info: SMIRInfo, start_symbol: str = 'main', depth: int | None = None) -> Pattern:
+        init_config, init_constraints = self.make_call_config(smir_info, start_symbol=start_symbol, init=True)
+        if len(free_vars(init_config)) > 0 or len(init_constraints) > 0:
+            raise ValueError(f'Cannot run function with variables: {start_symbol} - {free_vars(init_config)}')
```

Comment:
We _can_ run functions with arguments but only with the Haskell backend.
OK to leave as-is for now, though, anyway we would not want to keep running them with the legacy backend and will have to develop a `krun` with booster support (branch management client-side).